File: src/main/kotlin/magicbook/gtlitecore/api/recipe/property/PseudoMultiPropertyValues.java
package magicbook.gtlitecore.api.recipe.property;

import lombok.Getter;
import lombok.Setter;
import net.minecraft.block.state.IBlockState;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.Arrays;

@Getter
@Setter
public class PseudoMultiPropertyValues
{

    public ArrayList<IBlockState> validBlockStates;

    public final String blockGroupName;

    public PseudoMultiPropertyValues(@NotNull String blockGroupName,
                                     @NotNull IBlockState... validBlockStates)
    {
        this.validBlockStates = new ArrayList<>(Arrays.asList(validBlockStates));
        this.blockGroupName = blockGroupName;
    }

    public PseudoMultiPropertyValues(@NotNull String blockGroupName,
                                     @NotNull ArrayList<IBlockState> validBlockStates)
    {
        this.validBlockStates = validBlockStates;
        this.blockGroupName = blockGroupName;
    }

}

File: src/main/kotlin/gregtechlite/gtlitecore/api/recipe/property/value/PseudoMultiPropertyValues.kt
package gregtechlite.gtlitecore.api.recipe.property.value

import net.minecraft.block.state.IBlockState

class PseudoMultiPropertyValues(val blockGroupName: String,
                                val validBlockStates: List<IBlockState>)
{

    constructor(blockGroupName: String,
                vararg validBlockStates: IBlockState) : this(blockGroupName, listOf(*validBlockStates))

}